pkg/deviceplugin/server.go

```diff
@@ -81,7 +81,7 @@ func newServer(devType string,
 func (srv *server) GetDevicePluginOptions(ctx context.Context, empty *pluginapi.Empty) (*pluginapi.DevicePluginOptions, error) {
 	return &pluginapi.DevicePluginOptions{
 		PreStartRequired:                srv.preStartContainer != nil,
-		GetPreferredAllocationAvailable: false,
+		GetPreferredAllocationAvailable: srv.getPreferredAllocation != nil,
 	}, nil
```

Comment:
I'd rather put this struct creation into a private method with narrow interface like `func (srv *server) getDevicePluginOptions() *pluginapi.DevicePluginOptions` and use it here and also inside `registerWithKublet()`.
This way we wouldn't need to create and immediately throw away the context.

Then we can drop `options` from `registerWithKublet()`'s signature.